lib/charms/opensearch/v0/opensearch_backups.py

```diff
@@ -618,7 +627,7 @@ def _close_indices_if_needed(self, backup_id: int) -> Set[str]:
     def _restore(self, backup_id: int) -> Dict[str, Any]:
         """Runs the restore and processes the response."""
         backup_indices = self._list_backups().get(backup_id, {}).get("indices", {})
-        output = self._request(
+        output = self.charm.opensearch.request(
             "POST",
             f"_snapshot/{S3_REPOSITORY}/{backup_id.lower()}/_restore?wait_for_completion=true",
```

Comment:
backup_id is an integer.

Reply:
So, that is explicitly tested in our CI. It is a string [as stated here](https://github.com/canonical/opensearch-operator/blob/2/edge/actions.yaml#L61), as I understand it [from upstream](https://github.com/opensearch-project/OpenSearch/blob/98dbc4a31cce52e2553523f07ee25c04a71a62b3/plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/async/UploadRequest.java#L23). The annotation is wrong, I will fix it.